### DISCUSSION TITLE
The HUD display design

### DISCUSSION THREAD

**Discussion by Abang111** [cyberbotics/webots#6325] [General]
Hello,

I want to add a head-up display (HUD) to an existing car model, and the model I'm currently using is the Bmw X5 model inside webots, but it seems to be a fixed whole from which I can't change a part. 

My idea is to create the same windshield and fit it tightly onto the existing BMW model, and set the texture, which is constantly updated and controlled in the “controller” module. However, the PROTO model of BmwX5 needs to be modified, these PROTO files are complicated to modify.

Is my method feasible? Or is there a better, more scientific way? Is there any example.

**Comment by omichel** [comment]
Yes, it seems feasible. It shouldn't be too difficult to add a [Display](https://cyberbotics.com/doc/reference/display) device to a PROTO model.
